src/solvers/refinement/string_constraint_generator.h

```diff
@@ -26,33 +25,20 @@ class string_constraint_generatort
   // to the axiom list.
 
   string_constraint_generatort():
-    mode(ID_unknown), refined_string_type(char_type())
-  { }
+    mode(ID_unknown)
+  {
+    next_symbol_id=1;
```

Comment:
I'm not entirely sure, but I think this would actually reset the counter. Place the initialisation with the declaration.